srai/regionizers/base.py

```diff
@@ -0,0 +1,22 @@
+"""Base class for regionizers."""
+
+import abc
+from typing import Any
+
+import geopandas as gpd
+
+
+class BaseRegionizer(abc.ABC):
+    """Base abstract class for regionizers."""
+
+    @abc.abstractmethod
+    def transform(self, gdf: gpd.GeoDataFrame, **kwargs: Any) -> gpd.GeoDataFrame:
```

Comment:
Are we expecting the regionizer to union all polygons inside into a single multi polygon and regionize everything, or use just the first record? Not obvious from the description.
I think we shouldn't allow those two different ways to be used in different implementations.
Best way would be to union all those polygons (remove points) into a single geometry and ask regionizer to implement working on a single geometry (multi polygon) with **kwargs.

Reply:
And speaking about `gdf.index` as an ID of the record - I'd suggest adding an internal function that calculates a hash from the geometry WKT / WKB if no ID has been provided during regionizing (H3, GeoHash, OSM id, ...). Geometries of regions should be disjointed, so there won't be any geometries that are equal and will produce the same hash.